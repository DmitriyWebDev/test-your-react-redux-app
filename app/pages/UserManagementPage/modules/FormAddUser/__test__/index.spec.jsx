import React from 'react';
import { shallow } from 'enzyme';
import { FormAddUser } from '../index';
import expect from 'expect';

describe('FormAddUser component', () => {
  const defaultProps = {
    disabled: false,
    addUser: () => ({}),
  };

  it('should render correctly', () => {
    const component = shallow(<FormAddUser {...defaultProps} />);

    expect(component).toMatchSnapshot();
  });

  it('Changes firstName correctly', () => {
    const wrapper = shallow(<FormAddUser {...defaultProps} />);
    const newFirstName = 'newFirstName';

    wrapper.instance().handleFirstNameChange(newFirstName);

    expect(wrapper.state('firstName')).toBe(newFirstName);
  });
});
