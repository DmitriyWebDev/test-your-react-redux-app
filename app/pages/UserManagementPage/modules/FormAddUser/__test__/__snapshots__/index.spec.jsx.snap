// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormAddUser component should render correctly 1`] = `
<div
  className="form"
>
  <div
    className="form__title"
  >
    Users form
  </div>
  <div
    className="form__content"
  >
    <div
      className="form__field"
    >
      <div
        className="form__fieldLabel"
      >
        Name
      </div>
      <div
        className="form__fieldValue"
      >
        <Input
          className=""
          id=""
          name="firstName"
          onChange={[Function]}
          placeholder=""
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="form__field"
    >
      <div
        className="form__fieldLabel"
      >
        Surname
      </div>
      <div
        className="form__fieldValue"
      >
        <Input
          className=""
          id=""
          name="lastName"
          onChange={[Function]}
          placeholder=""
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="form__field"
    >
      <div
        className="form__fieldLabel"
      >
        Position
      </div>
      <div
        className="form__fieldValue"
      >
        <Input
          className=""
          id=""
          name="position"
          onChange={[Function]}
          placeholder=""
          type="text"
          value=""
        />
      </div>
    </div>
    <Button
      className=""
      disabled={true}
      id=""
      onClick={[Function]}
    >
      Add user
    </Button>
  </div>
</div>
`;
